include/circt/Dialect/SV/TypeDecl.td

```diff
@@ -0,0 +1,98 @@
+//===- TypeDecl.td - SV type declaration ops ---------------*- tablegen -*-===//
+//
+// This describes the ops for SystemVerilog type declartions.
+//
+//===----------------------------------------------------------------------===//
+
+def HasTypeDeclTerminator :
+    SingleBlockImplicitTerminator<"TypeDeclTerminatorOp">;
+
+//===----------------------------------------------------------------------===//
+// Interface operations
+//===----------------------------------------------------------------------===//
+
+def InterfaceOp : SVOp<"interface",
+    [Symbol, SymbolTable, HasTypeDeclTerminator]> {
+  let summary = "Operation to define a SystemVerilog interface";
+
+  let description = [{
+    This operation defines a named interface. Its name is a symbol that can
+    be looked up when declared inside a SymbolTable operation. This operation is
+    also a SymbolTable itself, so the symbols in its region can be looked up.
+  }];
```

Comment:
I think there's lots of value in having concrete dialect examples in the documentation.

Reply:
Good point, I will add some here.